packages/tizen_rpc_port/example/client/lib/message_client.dart

```diff
@@ -0,0 +1,211 @@
+    parcel.writeBool(once);
+  }
+
+  @override
+  void deserialize(Parcel parcel) {
+    id = parcel.readInt32();
+    sequenceId = parcel.readInt32();
+    once = parcel.readBool();
+  }
+}
+
+typedef NotifyCallback = void Function(String, String);
+
+class _NotifyCallback extends _CallbackBase {
+  _NotifyCallback(NotifyCallback callback, {bool once = false})
+      : super(_DelegateId.notifyCallback.id, once, callback);
+
+  @override
+  Future<void> _onReceivedEvent(Parcel parcel) async {
+    final String sender = parcel.readString();
+    final String message = parcel.readString();
+    callback?.call(sender, message);
+  }
+}
+
+typedef OnDisconnected = Future<void> Function();
+
+class Message extends ProxyBase {
+  Message(String appid) : super(appid, 'Message');
+
+  bool _isOnline = false;
+  final List<_CallbackBase> _delegateList = <_CallbackBase>[];
+
+  OnDisconnected? _onDisconnected;
+
+  @override
+  @nonVirtual
+  Future<void> onDisconnectedEvent() async {
+    await super.onDisconnectedEvent();
+    _isOnline = false;
+    await _onDisconnected?.call();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onReceivedEvent(Parcel parcel) async {
+    final int cmd = parcel.readInt32();
+    if (cmd != _MethodId.callback.id) {
+      return;
+    }
+
+    final int id = parcel.readInt32();
+    final int sequenceId = parcel.readInt32();
+    final bool once = parcel.readBool();
+    for (final _CallbackBase delegate in _delegateList) {
+      if (delegate.id == id && delegate.sequenceId == sequenceId) {
+        await delegate._onReceivedEvent(parcel);
+        if (delegate.once && once) {
+          _delegateList.remove(delegate);
+        }
+        break;
+      }
+    }
+  }
+
+  Parcel _consumeCommand(Port port) {
+    try {
+      final Parcel parcel = Parcel.fromPort(port);
+      final int cmd = parcel.readInt32();
+      if (cmd != _MethodId.result.id) {
+        throw Exception('Received parcel is invalid. $cmd');
+      }
+      return parcel;
+    } catch (error) {
+      return Parcel();
+    }
+  }
+
+  @override
+  Future<void> connect({OnDisconnected? onDisconnected}) async {
+    await super.connect();
+    _onDisconnected = onDisconnected;
+    _isOnline = true;
+  }
+
+  @override
+  Future<void> disconnect() async {
+    await super.disconnect();
+    _isOnline = false;
+  }
+
+  void disposeCallback(Function callback) {
+    _delegateList
+        .removeWhere((_CallbackBase element) => element.callback == callback);
+  }
+
+  Future<int> register(String name, NotifyCallback callback) async {
+    if (!_isOnline) {
+      throw StateError('Must be connected first');
+    }
+
+    final Port port = getPort(PortType.main);
+    final Parcel parcel = Parcel();
+    final ParcelHeader header = parcel.header;
+    header.tag = _tidlVersion;
+    parcel.writeInt32(_MethodId.register.id);
+    parcel.writeString(name);
+    final _NotifyCallback callbackBase = _NotifyCallback(callback);
```

Comment:
Thanks for the review!
I agree to change the name of `_CallbackBase` to `_Delegate`.
However, since `_NotifyCallback` is specified in the above part, it is not necessary to declare it as a base class.